File: go/sql_injection/sort_allowlist.go
package sqlinjection

// allowedSortColumns maps the sort field names accepted from API requests to
// the SQL column identifiers they resolve to.
var allowedSortColumns = map[string]string{
	"date":     "created_at",
	"total":    "total",
	"status":   "status",
	"customer": "user_id",
}

// SortColumn returns the SQL column mapped to a requested sort field. ok is
// false when the field is not in allowedSortColumns.
func SortColumn(field string) (column string, ok bool) {
	column, ok = allowedSortColumns[field]
	return column, ok
}

File: php/sql_injection/sort_allowlist.php
<?php
/**
 * Shopist - Sort Field Allowlist
 * DEMO FILE: Intentional false positive for the Datadog Code Security SAST demo.
 * DO NOT USE IN PRODUCTION.
 */

// resolveSortColumn returns the SQL column mapped to a requested sort field, or
// null when the field is not in the allowlist.
function resolveSortColumn($field) {
    $allowedSortColumns = [
        'date'     => 'created_at',
        'total'    => 'total',
        'status'   => 'status',
        'customer' => 'user_id',
    ];
    return $allowedSortColumns[$field] ?? null;
}
